server/src/parsers/odt.rs

```diff
@@ -0,0 +1,449 @@
+extern crate serde_json;
+extern crate xml;
+extern crate zip;
+
+use serde_json::map::Map;
+use serde_json::value::Value;
+use serde_json::Number;
+use std::fs;
+use xml::reader::{EventReader, XmlEvent};
+
+pub struct ODTParser {
+    body_begin: bool,
+    styles_begin: bool,
+    auto_styles: Map<String, Value>,
+    is_span: bool,
+    current_span_style: String,
+    set_children_underline: bool,
+    ensure_children_no_underline: bool,
+    document_hierarchy: Vec<Value>,
+    archive: zip::ZipArchive<std::fs::File>,
+}
+
+impl ODTParser {
+    /// Initialises the struct's members given the path of an ODT file,
+    /// can return an error if there is an issue with the file
+    pub fn new(filepath: &str) -> Result<ODTParser, String> {
+        let archive = get_archive(filepath);
+        if let Err(e) = archive {
+            return Err(e);
+        }
+        let archive = archive.unwrap();
+
+        let mut document_contents: Map<String, Value> = Map::new(); //value of the "document" key
+        document_contents.insert(
+            "title".to_string(),
+            Value::String("Kauri (Working title)".to_string()),
+        );
+        document_contents.insert("paper".to_string(), Value::String("A4".to_string()));
+        document_contents.insert("children".to_string(), Value::Array(Vec::new()));
+
+        let mut document_hierarchy: Vec<Value> = Vec::new(); //in case of nested tags, not actually handled yet
```

Comment:
Consistency with Rust style RFC

> Prefer to put a comment on its own line.

```suggestion
        // in case of nested tags, not actually handled yet
        let mut document_hierarchy: Vec<Value> = Vec::new();
```